dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -44,8 +44,10 @@ internal class DatadogDataConstraints : DataConstraints {
         val convertedAttributes = attributes.mapNotNull {
             // We need this in case the attributes are added from JAVA code and a null key may be
             // passed.
+            @Suppress("SENSELESS_COMPARISON")
             if (it.key == null) {
                 devLogger.e("\"$it\" is an invalid attribute, and was ignored.")
+                @Suppress("UNUSED_EXPRESSION")
                 null
```

Comment:
This warning is actually legitimate, it seems across the list of changes, we changed how this method behaves.
originally this `mapNotNull` would convert attributes to null if the key was null. We lost this behavior back in november it seems.

Reply:
Hmm...the way I understand this code is:
1. We add this extra check because as it is explained above in the comments this can be called from Java and that key can actually be null.
2. In case we discover a null key, we return null which will be handled by `mapNotNull` function by discarding it. Am I missing something ?

Reply:
No that's exactly it